### PR TITLE
Add 2 fuzzers

### PR DESCRIPTION
**Pull request by AdamKorcz**
There have been discussions on adding fuzzing to Istio and setting up continuous fuzzing as well. See https://github.com/istio/istio/issues/14935

This PR adds 2 fuzzers that are implemented by way of Dvyukovs go-fuzz which is supported by OSS-fuzz.

In a few minutes I will set up the integration files on the OSS-fuzz side. They will fail for now, but once the fuzzers are merged here, they will build sucessfully.

### CODE REVIEW

**Comment by AdamKorcz** [comment]
I have set up the integration on the OSS-fuzz side here: https://github.com/google/oss-fuzz/pull/4762 

As mentioned above, once the fuzzers in this PR are merged and at least one maintainers email address is added in the `project.yaml` file in the PR on OSS-fuzz, the integration is ready for the OSS-fuzz maintainers to review.

**Comment by AdamKorcz** [comment]
Pinging @howardjohn

File: tests/fuzz/fuzz.go
Comment (howardjohn) [inline tests/fuzz/fuzz.go:33]: What is the "cost" of each of these? For example, a unit test like this would run in microseconds, so even if its not super useful its not a high cost. Is this going to be eating up substantial CPU/time/$$?

I am asking in general to get a feel since I am not super familiar with fuzzing, but also this function is only used at build time so its probably low priority.

**Review by howardjohn — COMMENTED** [review]
Neat, thanks for working on this! I think this will become super useful if/when we can also fuzz complex data types (ie Istio/Kubernetes objects) and run them through large parts of our code

**Review by howardjohn — APPROVED** [review]

File: tests/fuzz/fuzz.go
Comment (howardjohn) [inline tests/fuzz/fuzz.go:20]: should be `istio.io/istio`

**Review by howardjohn — CHANGES_REQUESTED** [review]

File: tests/fuzz/fuzz.go
Comment (AdamKorcz) [reply tests/fuzz/fuzz.go:33]: The cost of an execution varies from target to target. Some applications require lots of initialization and executes less than 10 times per second. Other fuzzers execute 100.000 times per second. In the case of the ParseAndBuild-fuzzer, it has about 8000 executions per second on my local machine.

File: tests/fuzz/fuzz.go
Comment (AdamKorcz) [reply tests/fuzz/fuzz.go:20]: In the current build `istio.io/istio` results in "cannot find package..." for both imports.

Comment (howardjohn) [reply tests/fuzz/fuzz.go:20]: With this code running `go-fuzz-build` fails complaining about the import for me.

Comment (AdamKorcz) [reply tests/fuzz/fuzz.go:20]: Aha. I am running this through OSS-fuzz and there might be a bit of difference there. Do you prefer the fuzzers to be compile'able with go-fuzz-build before merging? It could also be a solution to make a rule in the `Makefile` that compiles the fuzzers

Comment (howardjohn) [reply tests/fuzz/fuzz.go:20]: It would be useful to have some way to run it locally, whether its through `go-fuzz-build` or otherwise, since it will be useful for debugging/dev

Comment (AdamKorcz) [reply tests/fuzz/fuzz.go:20]: Sure, no problem. Did you get istio with `go get github.com/istio/istio/...` like in the Dockerfile?

Comment (howardjohn) [reply tests/fuzz/fuzz.go:20]: No, I have it cloned into `go/src/istio.io/istio`. I don't think it will build if you have it under github due to how go building works?

Comment (AdamKorcz) [reply tests/fuzz/fuzz.go:20]: I see the issue locally. I will work on a way to have it run locally with `go-fuzz-build` as well.

Comment (AdamKorcz) [reply tests/fuzz/fuzz.go]: I have updated the import paths.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**
